agreementbot/secure_api.go

```diff
@@ -591,9 +592,9 @@ func (a *SecureAPI) secrets(w http.ResponseWriter, r *http.Request) {
 		if vaultSecretName == "" {
 			secretNames, err := a.secretProvider.ListOrgSecrets(ec.GetExchangeId(), ec.GetExchangeToken(), org)
 
-			if err != nil {
-				glog.Errorf(APIlogString(fmt.Sprintf("unable to access secrets provider, error: %v.", err)))
-				writeResponse(w, msgPrinter.Sprintf("unable to access secrets provider, error: %v.", err), http.StatusInternalServerError)
+			if serr, ok := err.(secrets.ErrorResponse); err != nil && ok {
```

Comment:
Should it be serr != nil ?

Reply:
Yes, it fails when err is nil. There is a test added in `agbot_apitest.sh` which covers that case. As for `serr!=nil` I don't think it matters since the flag `ok` should indicate a failed conversion whether the error is nil or some incompatible type. Anyways, I should change it to `serr!=nil` since that makes more sense in this context.